src/js/helpers/offsetCalculator.js

```diff
@@ -10,9 +10,18 @@
 
 import getOffset from './../libs/offset';
 import getInlineOption from './getInlineOption';
+import validContainer from './container';
 
-export const getPositionIn = (el, defaultOffset, defaultAnchorPlacement) => {
-  const windowHeight = window.innerHeight;
+export const getPositionIn = (
+  el,
+  defaultOffset,
+  defaultAnchorPlacement,
+  container
+) => {
+  const windowHeight =
+    container === window
+      ? validContainer(container).innerHeight
+      : validContainer(container).clientHeight;
```

Comment:
What do you think about creating “getContainer()” function in container.js that validates the container and returns an object with top offset and height, being a universal wrapper that allows us to not care about clientHeight or innerHeight here? We could then just use getContainer(container).topOffset

Or alternatively: getOffset(container) and getHeight(container)

Reply:
This seems like overengineering for where a small refactor would be enough.

- Why is there argument validation in an helper ? Shouldn't helpers be generic 
? This seems required because `getPositionIn` is called with `options.container` because the `prepare` helper only receive `options`. We should have the container validated and resolved in the main function and then passed in to all helpers.

- If we want some abstraction to deal with `innerHeight` and `clientHeight`, this have nothing to do with `container` resolution. I don't think that's really useful, but the best way to do it would be a generic `getElementHeight` helper.

Reply:
> We should have the container validated and resolved in the main function and then passed in to all helpers.

If the helpers API looks weird with `container` in, then they are not generic helpers and there should be regrouped with the main function in a single `AOS` class.

An other solution (much easier but not really cleaner) would be to replace `option.container` with the resolved container.

Reply:
Great idea, will do

Reply:
I agree that passing resolved container + using generic helper for getting height would be the cleanest solution here. Thanks for an additional insight @ncoden 

I don't see a problem with having a generic helper that takes `Element` as an argument.